File: structural/flyweight/typescript/context.ts
import { Circle } from "./concrete-flyweight";

export class CircleContext {
    private circle: Circle;
    private x: number;
    private y: number;
    private radius: number;

    constructor(circle: Circle, x: number, y: number, radius: number) {
        this.circle = circle;
        this.x = x;
        this.y = y;
        this.radius = radius;
    }

    draw(): void {
        this.circle.draw(this.x, this.y, this.radius);
    }
}

File: structural/flyweight/python/context.py
from concrete_flyweight import Circle


class CircleContext:
    def __init__(self, circle: Circle, x: int, y: int, radius: int) -> None:
        self.circle = circle
        self.x = x
        self.y = y
        self.radius = radius

    def draw(self) -> None:
        self.circle.draw(self.x, self.y, self.radius)
